please_go tool panics when no version is supplied

I noticed this issue with the tool. It will panic on my mac when no version is supplied. Is this expected?

```
➜  core git:(refactor) ✗ plz run ///go//tools/please_go -- get github.com/redis/go-redis/v9       
Upgrading Please from version 17.0.0 to 17.8.0
panic: runtime error: index out of range [1] with length 1

goroutine 1 [running]:
github.com/please-build/go-rules/tools/please_go/goget.(*getter).goGet(0x1400018fed8?, {0x140001ec940, 0x1, 0x6?})
   tools/please_go/goget/go_get.go:112 +0x26c
github.com/please-build/go-rules/tools/please_go/goget.GoGet(...)
   tools/please_go/goget/go_get.go:130
main.glob..func8()
   tools/please_go/please_go.go:200 +0xd0
main.main()
   tools/please_go/please_go.go:244 +0x84
➜  core git:(refactor) ✗ plz run ///go//tools/please_go -- get github.com/redis/go-redis/v9@v9.5.1
Upgrading Please from version 17.0.0 to 17.8.0
go_repo(module="github.com/dgryski/go-rendezvous", version="v0.0.0-20200823014737-9f7001d12a5f")
go_repo(module="github.com/redis/go-redis/v9", version="v9.5.1")
go_repo(module="github.com/bsm/ginkgo/v2", version="v2.12.0")
go_repo(module="github.com/bsm/gomega", version="v1.27.10")
go_repo(module="github.com/cespare/xxhash/v2", version="v2.2.0")
```

--- a/tools/please_go/goget/go_get.go
+++ b/tools/please_go/goget/go_get.go
@@ -1,8 +1,10 @@
 package goget
 
 import (
+	"errors"
 	"fmt"
 	"io"
+	"log"
 	"net/http"
 	"os"
 	"strings"
@@ -75,8 +77,7 @@ func (g *getter) getGoModWithFallback(mod, version string) (*modfile.File, error
 	for mod, version := range modVersionsToAttempt {
 		modFile, err := g.getGoMod(mod, version)
 		if err != nil {
-			// TODO: when we upgrade to Go 1.20, use `errors.Join(...)`.
-			errs = fmt.Errorf("%w: %w", errs, err)
+			errs = errors.Join(errs, err)
 			continue
 		}
 
@@ -108,8 +109,11 @@ func (g *getter) getDeps(deps map[string]string, mod, version string) error {
 func (g *getter) goGet(mods []string) error {
 	deps := map[string]string{}
 	for _, mod := range mods {
-		parts := strings.Split(mod, "@")
-		deps[parts[0]] = parts[1]
+		path, version, ok := strings.Cut(mod, "@")
+		if !ok {
+			log.Fatalf("Module spec %s is missing a version; must be in the format golang.org/x/sys@v1.0.0", mod)
+		}
+		deps[path] = version
 	}
 
 	for mod, ver := range deps {

